### DISCUSSION TITLE
Calculating an indicator such as a EMA on the High rather than the close

### DISCUSSION THREAD

**Discussion by theodorebeodore** [DaveSkender/Stock.Indicators#1053] [Help and support]
Hi 
Sorry if this is very simplistic but I have been trying to calculate the EMA of the High using the standard Quote class.  Even if someone can point me in the direction of the appropriate documentation that would be awesome.

**Comment by DaveSkender** [comment]
The helper [.Use()](https://dotnet.stockindicators.dev/utilities/#use-alternate-price) function will do what you’re seeking.

```csharp
var results = quotes
  .Use(CandlePart.High)
  .GetEma(14);
```

**Reply by theodorebeodore** [reply]
thanks  for the quick reply!

**Reply by theodorebeodore** [reply]
var results = quotes
            .Use(CandlePart.HL2)
            .GetEma(8 *2-1);
        var V1 = results.FirstOrDefault(x => x.Date == quotesList[t].Date);
        Console.WriteLine($"{V1.Date} {V1.Ema}");

With the above code. i am trying to get the Ema to print to confirm but the V1.Ema is coming up blank.  CandlePart.HL2 i am assuming calculates the (high+Low)/2

**Reply by DaveSkender** [reply]
Yes, your syntax is correct; however [EMA](https://dotnet.stockindicators.dev/indicators/Ema/#content) requires a warmup before it is accurate, due to [convergence](https://github.com/DaveSkender/Stock.Indicators/discussions/688).  For EMA, the first N-1 periods will have null values since there’s not enough data to calculate; and the first N+100 periods will have convergence-related precision errors.  Try using 150 or more historical quotes when N=15.

**Reply by theodorebeodore** [reply]
OHHH man! thanks sorry I feel stupid for asking that. Works perfectly when looping all the quotes. I need to take a coding break :) 
Thanks again!
